php/libraries/DefaultFilesUploadHandler.class.inc

```diff
@@ -0,0 +1,367 @@
+     * array.
+     *
+     * @return array array of all the errors found.
+     */
+    public function getErrors() : array
+    {
+        return $this->_errors;
+    }
+
+    /**
+     * Accessor for field $_lorisForm.
+     *
+     * @return LorisForm the LorisForm associated with this handler.
+     */
+    public function getLorisForm() : \LorisForm
+    {
+        return $this->_lorisForm;
+    }
+
+    /**
+     * Accessor for field $_dirPermissions.
+     *
+     * @return int value for field $_dirPermissions.
+     */
+    public function getDirPermissions() : int
+    {
+        return $this->_dirPermissions;
+    }
+
+    /**
+     * Accessor for field $_moveDirPath.
+     *
+     * @return string value for field $_moveDirPath.
+     */
+    public function getMoveDirPath() : string
+    {
+        return $this->_moveDirPath;
+    }
+
+    /**
+     * Accessor for field $_dirGroup.
+     *
+     * @return int value for field $_dirGroup.
+     */
+    public function getDirGroup() : string
+    {
+        return $this->_dirGroup;
+    }
+
+    /**
+     * Accessor for field $_movedFiles.
+     *
+     * @return array the final location of all files uploaded on the
+     *               server. Each key is the name of a LorisFirm field
+     *               used to upload a file and its corresponding value is the final
+     *               location for the uploaded file on the server.
+     */
+    public function getMovedFiles() : array
+    {
+        return $this->_movedFiles;
+    }
+
+    /**
+     * Accessor for field $_dirPermissions.
+     *
+     * @return int value for field $_dirPermissions.
+     */
+    public function getFiles() : array
+    {
+        if (is_null($this->_files)) {
+            $this->_files = array();
+            foreach ($this->_lorisForm->getElements() as $element) {
+                if ($element instanceof LorisFormFileElement) {
+                    $this->_files[$element->getName()] = $element->getValue();
+                }
+            }
+        }
+
+        return $this->_files;
+    }
+
+    /**
+     * Determines if the set of uploaded files is valid or not, updating the $_errors
+     * array with the errors found. This method will look at all the fields of the
+     * LorisForm associated to this handler and verify that the upload error code
+     * found in the super global $_FILES for that field is UPLOAD_ERR_OK
+     * (see documentation for $_FILES). If no file name was entered for a given field
+     * in the LorisForm, then the upload is considered invalid and the error for that
+     * field will be set to false. This method assumes that all the fields used for
+     * file uploads have a value (i.e. no "optional" upload is allowed). This
+     * behavior can be overridden bu subclasses should they need to.
+     *
+     * @return bool true if the uploaded file set is valid, false otherwise.
+     */
+    public function isValid() : bool
```

Comment:
The `isValid` function name is misleading. There is no validation of the file(s). This is checking if the upload process complete successfully. `isUploadSuccess` or `isUploadOK` would be more accurate in my opinion